### DISCUSSION TITLE
Why must I restart the master container after every daily backup for web login to work?

### DISCUSSION THREAD

**Discussion by maste9** [nextcloud/all-in-one#4009] [Questions] [answered]
I have activated daily backup which works nice in general. Also I don't receive any errors after backup and automated container restarts from my diverse clients.

Except:
My open browser session is logged off. And when I try to login again I can't because the **login screen loops always back** and informs about an _temporary failure_ and that i should try it again. But it doesn't matter how often I try.

To rule out that it has to do with any session id or likewise I opened a **private window in Firefox** to login but received the same error message. The same with another account that hadn't been logged in on that browser.

After I ```docker-compose restart``` the master container everything works **fine again**. But that's of course not the plan I guess ;) 

Anyone else experiencing that issue or even an idea about how to fix this?

Thanks in advance!

**Comment by szaimen** [comment]
Hi, this sounds like https://github.com/nextcloud/server/issues/32538

**Comment by maste9** [comment]
I fear it's not. I took a look into the current log while it's happening again.

It's the u2f provider that fails to load 🤔  `[core] Fehler: 1 two-factor auth providers failed to load`
> {"reqId":"aEaFsW3GELLrmVGy1QEK","level":3,"time":"2024-01-09T12:34:18+00:00","remoteAddr":"89.245.xxx.xxx","user":"redacted","app":"core","method":"POST","url":"/login/challenge/totp","message":"1 two-factor auth providers failed to load","userAgent":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:121.0) Gecko/20100101 Firefox/121.0","version":"27.1.5.1","data":{"app":"core"},"id":"659d3d5eded5a"}

I try to look into that further.

## Update
OK, I had not only *Two-Factor TOTP Provider* installed but also _Two-Factor WebAuthn_ at once and they seem to have interfered. After deactivating the second I could login again.
I will observe that further but hopefully it's fixed now.

**Reply by maste9** [reply]
Fuuu ... No, unfortunately that also didn't fix it. Still the same problem and the TOTP Provider fails to start now.
```
{"reqId":"BVTjIEQMsfFwrQzm4q5L","level":3,"time":"2024-01-12T07:29:50+00:00","remoteAddr":"146.60.x.x","user":"redacted","app":"core","method":"POST","url":"/login/challenge/totp","message":"2 two-factor auth providers failed to load","userAgent":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:121.0) Gecko/20100101 Firefox/121.0","version":"27.1.5.1","data":{"app":"core"},"id":"65a0ea95d5a3b"}
```

Edit: Interestingly it also tries to still load the webauth app which I have already deactived _and_ deleted. Maybe that is due to the migration I did the [hard way](https://github.com/nextcloud/all-in-one/blob/main/migration.md#migrate-the-files-and-the-database)? Yes I followed the instructions installing all apps before restore of database and had them updated before backup.

**Reply by szaimen** [reply]
Hm, can you disable the webauth correclty by enabling and disabling it another time from the webinterface?

**Reply by maste9** [reply]
I can, I did, I get still the same result unfortunately. 

I could login via private browser window immediately (using TOTP) and activate/deactivate. But trying to login at the current (non-private) firefox window led to nothing but bruteforce protection after a bunch of trials. After more than thirty seconds it still says I'm locked. Trying to login leads again to failure. Except when I login via private window again 🤷 
Whitelisting my IP, login again  - no, temporary failure.
Restarting master-container via portainer - no, failure again.
From Private Window trying to open the AIO interface to down and up the containers:
```
The login is blocked since Nextcloud is running.
Please use the [automatic login](https://github.com/nextcloud/all-in-one#how-to-easily-log-in-to-the-aio-interface) from your Nextcloud.

If that is not possible, you can unblock the login by running
sudo docker stop nextcloud-aio-apache
```
Done so, restarted it even if it's not instructed ... No. Still the same message when I try to open the interface ......
Trying to stop the containers handish via portainer, yes works, ...
<details><summary>Details</summary>
<p>
 downing the master-container, no. Because npm is still in the network as  I had to put it into the aio's network to prevent bruteforce protection from blocking my proxy  (npm)'s internal ip that might change at update. But that networks stuff is probably obsolete now as I had that problem trying to get UnifiedPush Provider-App to work which I gave a kick lately as it's obviously impossible and hosting ntfy server now.
</p>
</details> 
... returning to different networks and at last starting the containers again by interface I can reach now 
<details><summary>Details</summary>
<p>

 (by hand in the non-private browser window; the impossibility of reaching interface might have been due to calling it in a private window because that's still happending when I call interface there) ...

</p>
</details>
Aaaand I can login. 

Sorry for that trial and error report but there's a chance you will see my mistake or what ever's interfering here in my setup. Don't want to miss out the important detail I might oversee.

Half an hour later, switched device: Same procedure. Open browser, open Nextcloud Tab, not longer logged in. Try log login Temporary failure and doesn't matter how often I try, if I close tab and open a new one ...
Private Window: Failure once, and on second attempt I can login.
Well, there's obviously a pattern but I know far too little about the under the hood processes to understand what might causing it.

**Reply by maste9** [reply]
It seems to be solved in the meantime. At least I hadn't any further problems logging in in the last weeks and stay logged in even after the containers restart during the night for backups.

So I close this hopefully :crossed_fingers:

**Comment by maste9** [comment, marked as answer]
It seems to be solved in the meantime. At least I hadn't any further problems logging in in the last weeks and stay logged in even after the containers restart during the night for backups.

So I close this hopefully :crossed_fingers:
